packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx

```diff
@@ -0,0 +1,285 @@
+The component will rely on data accessed via our utility hooks:
+
+- [`useCall`](../../call-engine/hooks-and-contexts#usecall)
+- [`useCallCallingState`](../../call-engine/hooks-and-contexts#usecallcallingstate)
+- [`useCallMetadata`](../../call-engine/hooks-and-contexts#usecallmetadata)
+- [`useCallMembers`](../../call-engine/hooks-and-contexts#usecallmembers)
+- [`useMediaDevices`](../../call-engine/hooks-and-contexts#usemediadevices)
+
+### User preview
+
+The panel will render [`VideoPreview`](todo docs link) so that the user can see the camera input or avatars if video preview is disabled. The avatar display logic is as follows:
+- show all the called users' avatars if the call is outgoing (who am I calling)
+- show the avatar of the person who initiated the call in case of an incoming call (who is calling me)
+
+:::note
+Note, how the initial video state is handled by [`MediaDevicesProvider`](../../call-engine/hooks-and-contexts#mediadevicesprovider). We only consume the API through corresponding context consumer hook [`useMediaDevices`](../../call-engine/hooks-and-contexts#usemediadevices).
+:::
+```tsx
+import {
+  useCall,
+  useCallMembers,
+  useCallMetadata,
+  useMediaDevices,
+} from '@stream-io/video-react-sdk';
+
+type PendingCallPanelProps = {
+  showMemberCount?: number;
+};
+
+export const CustomPendingCallPanel = ({
+  showMemberCount = 3,
+}: PendingCallPanelProps) => {
+  const call = useCall();
+  const metadata = useCallMetadata();
+  const members = useCallMembers();
+  // highlight-next-line
+  const { initialVideoState } = useMediaDevices();
+
+  if (!call) return null;
+
+  const caller = metadata?.created_by;
+  // show the caller if this is an incoming call or show all the users I am calling to
+  const membersToShow = call.isCreatedByMe
+    ? members
+        ?.slice(0, showMemberCount)
+        .map(({ user }) => user)
+        .filter((u) => !!u) || []
+    : caller
+    ? [caller]
+    : [];
+
+  return (
+    <div className="str-video__call-panel str-video__call-panel--pending">
+      // highlight-next-line
+      {initialVideoState.enabled ? (
+        <VideoPreview />
+      ) : (
+        <CallMembers members={membersToShow} />
+      )}
+      <CallCallingStateLabel />
+      <CallControls />
+    </div>
+  );
+};
+```
+
+#### Video preview enablement
+
+Enable the initial video in [`VideoPreview`](todo docs link) component with `mediaDevicesProviderProps` prop on [`StreamCall`](todo docs link) component as follows:
+
+```tsx
+import { StreamCall, useCalls } from '@stream-io/video-react-sdk';
+
+import { CallPanel } from './CallPanel';
+
+export const Video = () => {
+  const calls = useCalls();
+  return (
+    <>
+      {calls.map((call) => (
+        <StreamCall
+          call={call}
+          autoJoin={false}
+          key={call.cid}
+          // highlight-next-line
+          mediaDevicesProviderProps={{ initialVideoEnabled: true }}
+        >
+          <CallPanel />
```

Comment:
Isn't this component called `CustomPendingCallPanel` in the prev code snippet?

Reply:
No, but it is true, that I am not explaining that the CustomPendingCallPanel is a child of CallPanel. I will add that information.